### PR TITLE
Allow to disable linkerd-viz namespace installation with Helm

### PR DESCRIPTION
**Pull request by nlamirault**
Signed-off-by: Nicolas Lamirault <nicolas.lamirault@gmail.com>

<!--  Thanks for sending a pull request!

If you already have a well-structured git commit message, chances are GitHub
set the title and description of this PR to the git commit message subject and
body, respectively. If so, you may delete these instructions and submit your PR.

If this is your first time, please read our contributor guide:
https://github.com/linkerd/linkerd2/blob/main/CONTRIBUTING.md

The title and description of your Pull Request should match the git commit
subject and body, respectively. Git commit messages are structured as follows:

``

Solution

Validation

Fixes #[GitHub issue ID]

DCO Sign off
```

Example git commit message:

```
Introduce Pull Request Template

GitHub's community guidelines recommend a pull request template, the repo was
lacking one.

Introduce a `PULL_REQUEST_TEMPLATE.md` file.

Once merged, the
[Community profile checklist](https://github.com/linkerd/linkerd2/community)
should indicate the repo now provides a pull request template.

Fixes #3321

Signed-off-by: Jane Smith <jane.smith@example.com>
```

Note the git commit message subject becomes the pull request title.

For more details around git commits, see the section on Committing in our
contributor guide:
https://github.com/linkerd/linkerd2/blob/main/CONTRIBUTING.md#committing
-->

If namespace is manage by an external tool , it fails on install. 
Add a feature to not manage namespace by Helm.

### CODE REVIEW

**Review by alpeb — APPROVED** [review]
Good catch, this looks good to me :+1:

File: viz/charts/linkerd-viz/templates/namespace.yaml [hunk 1]
```diff
@@ -11,3 +12,4 @@ metadata:
     {{.Values.extensionAnnotation}}: linkerd-viz
   annotations:
     {{.Values.proxyInjectAnnotation}}: enabled
+{{ end -}}
```
Comment (kleimkuhler) [inline viz/charts/linkerd-viz/templates/namespace.yaml:15]: Can you please add a newline here?

File: viz/charts/linkerd-viz/README.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ Kubernetes: `>=1.13.0-0`
 | grafana.resources.memory.request | string | `nil` | Amount of memory that the grafana container requests |
 | identityTrustDomain | string | `"cluster.local"` | Trust domain used for identity  |
 | imagePullSecrets | list | `[]` | For Private docker registries, authentication is needed.  Registry secrets are applied to the respective service accounts |
+| installNamespace | bool | `true` | Create the namespace or not |
```
Comment (kleimkuhler) [inline viz/charts/linkerd-viz/README.md:102]: Small nit but can we use wording similar to the [base chart](https://github.com/linkerd/linkerd2/blob/2087c95dd8cec0b11e6ab04d1abf4a4019410977/charts/linkerd2/README.md#L200)?

```suggestion
| installNamespace | bool | `true` | Set to false when installing in a custom namespace. |
```

File: viz/charts/linkerd-viz/values.yaml [hunk 3]
```diff
@@ -30,6 +30,8 @@ createdByAnnotation: linkerd.io/created-by
 proxyInjectAnnotation: linkerd.io/inject
 extensionAnnotation: linkerd.io/extension
 
+# -- Create the namespace or not
```
Comment (kleimkuhler) [inline viz/charts/linkerd-viz/values.yaml:33]: And then same comment change here.

**Review by kleimkuhler — COMMENTED** [review]
Thanks @nlamirault! Left a few comments.

**Review by kleimkuhler — APPROVED** [review]
Thanks @nlamirault 👍

File: viz/charts/linkerd-viz/README.md (hunk 2 above)
Comment (Pothulapati) [reply viz/charts/linkerd-viz/README.md]: Very small nit, They can still set this true  even for custom namespace usecase too right? where they would set `namespace: xyz`

Comment (alpeb) [reply viz/charts/linkerd-viz/README.md]: Right,  the message can be a little misleading. Something that can be addressed in the upcoming Helm values cleanup work :wink:

Comment (kleimkuhler) [reply viz/charts/linkerd-viz/README.md]: Ah right sorry, that's my fault. Sounds good about fixing it.

**Pull request merged**